subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/DefaultCopySpecCodec.kt

```diff
@@ -74,16 +78,16 @@ class DefaultCopySpecCodec(
             val fileMode = readNullableSmallInt()
             val actions = readList().uncheckedCast<List<Action<FileCopyDetails>>>()
             val children = readList().uncheckedCast<List<CopySpecInternal>>()
-            val copySpec = DefaultCopySpec(fileCollectionFactory, instantiator, patternSetFactory, destPath, sourceFiles, patterns, actions, children)
+            val copySpec = DefaultCopySpec(fileCollectionFactory, objectFactory, instantiator, patternSetFactory, destPath, sourceFiles, patterns, actions, children)
             copySpec.duplicatesStrategy = duplicatesStrategy
             copySpec.includeEmptyDirs = includeEmptyDirs
             copySpec.isCaseSensitive = isCaseSensitive
             copySpec.filteringCharset = filteringCharset
             if (dirMode != null) {
-                copySpec.dirMode = dirMode
+                copySpec.dirPermissions.set(fileSystemOperations.permissions(Integer.toOctalString(dirMode)))
             }
             if (fileMode != null) {
-                copySpec.fileMode = fileMode
+                copySpec.filePermissions.set(fileSystemOperations.permissions(Integer.toOctalString(fileMode)))
```

Comment:
❓ Should we have a public API to set the permissions directly using an integer? It seems that we are now converting to-and-from a `String` to go in between.

Reply:
yes